### ISSUE TITLE
GDAL_DISABLE_READDIR_ON_OPEN default configuration is confusing for remote resources

### ISSUE THREAD

**Issue by hobu** [OSGeo/gdal#9443]
### Feature description

What is the case for defaulting to `GDAL_DISABLE_READDIR_ON_OPEN=FALSE` for remote VSI sources? Users frequently trip up over this feature, it's double-negative nature makes it very difficult to read, and you almost never want it in the use of drivers that are single file types. 

I wonder if the option should be renamed to `GDAL_ENABLE_READDIR_SCAN_ON_OPEN=TRUE`. Do other users have trouble with this option?

**Comment by hobu** [comment]
> This setting is fundamentally a GDAL core mechanism, implemented in the GDALOpenInfo class. Its general default value is NO.

This core GDAL mechanism was first written when people were assuming they were reading data from spinning disks. We almost never do that nowadays. 

I think the default should be `GDAL_DISABLE_READDIR_ON_OPEN=EMPTY_DIR` at the very least. I opened the ticket to discover if there are others with similar feelings. I do know that any cloud-oriented GDAL processing scripts are almost always littered with configuration management stuff to make sure to turn this behavior off, often after being burned by it.

**Comment by rouault** [comment]
I agree the computing environment has shifted a bit in the last 20 years :-) and the current situation is not ideal.

> I think the default should be `GDAL_DISABLE_READDIR_ON_OPEN=EMPTY_DIR` at the very least.

I don't think we can do that without a major break in backwards compatibility. That would break a lot of drivers that require or can make use of side car files. For example if users rely on external .ovr overviews.

**Comment by mdsumner** [comment]
Does this affect scanning a directory within an archive? (vsizip,vsitar, etc)  I've assumed it does, and is why I won't set this value as a default even though I think about doing that often ...

**Comment by rouault** [comment]
> Does this affect scanning a directory within an archive? (vsizip,vsitar, etc)

yes, it does. It controls whether directory listing is done. For /vsizip/, directory listing should be relatively fast (as the list of files is at the end of the ZIP). For /vsitar/, it will need to seek from file to file, as there is no file index
